File: Itinere_3/Laravel/demo/app/Http/Controllers/ProductController.php
<?php

namespace App\Http\Controllers;

use App\Models\Product;
use Illuminate\Http\Request;

class ProductController extends Controller
{
    public function read()
    {
        return view('read', ['products' => Product::all()]);
    }

    public function create(Request $request)
    {
        Product::create($request->all());
        return redirect('/read');
    }

    public function form(Request $request)
    {
        $product = Product::find($request->input('id'));

        if ($request->input("action") === "Modifica")
            return view('update', ['product' => $product]);

        if ($request->input("action") === "Rimuovi")
            $product->delete();

        return redirect('/read');
    }

    public function update(Request $request)
    {
        $product = Product::find($request->input('id'));
        $product->update($request->all()); // Simile alla create
        return redirect('/read');
    }
}

File: Itinere_3/Spring-Boot/Magazine/src/main/java/com/example/Magazine/controllers/ProductController.java
package com.example.Magazine.controllers;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;

import com.example.Magazine.models.Product;
import com.example.Magazine.repositories.ProductRepository;

@Controller
public class ProductController {
    private final ProductRepository repo;

    public ProductController(ProductRepository repo) {
        this.repo = repo;
    }

    @GetMapping("/")
    public String index(Model model) {
        model.addAttribute("products", repo.findAll());
        return "index";
    }

    @GetMapping("/create")
    public String create() {
        return "create";
    }

    @PostMapping("/store")
    public String store(Product product) {
        repo.save(product);
        return "redirect:/";
    }

    @GetMapping("/show")
    public String show(Long id, Model model) {
        model.addAttribute("product", repo.findById(id).orElse(null));
        return "show";
    }

    @GetMapping("/edit")
    public String edit(Long id, Model model) {
        model.addAttribute("product", repo.findById(id).orElse(null));
        return "edit";
    }

    @PostMapping("/update")
    public String update(Product product) {
        repo.save(product);
        return "redirect:/";
    }

    @PostMapping("/delete")
    public String delete(Long id) {
        repo.deleteById(id);
        return "redirect:/";
    }

    @GetMapping("/filter")
    public String filter(Model model, String action) {
        if (action.equals("filter")){
            model.addAttribute("products", repo.findByGiacenzaGreaterThan(0));
            return "filter";
        } else {
            return "redirect:/";
        }
    }

    @GetMapping("/decreaseStock")
    public String decreaseStock(Long id, Model model) {
        repo.decreaseStock(id);
        return show(id, model);
    }

    @PostMapping("/decreaseAllPrices")
    public String decreaseAllPrices(Integer percentage) {
        repo.decreaseAllPrices(percentage);
        return "redirect:/";
    }
}
